arm/Microsoft.RecoveryServices/vaults/deploy.bicep

```diff
@@ -131,33 +124,47 @@ resource rsv 'Microsoft.RecoveryServices/vaults@2021-08-01' = {
     tier: 'Standard'
   }
   properties: {}
+}
 
-  resource rsv_vaultstorageconfig 'backupstorageconfig@2020-02-02' = {
-    name: 'vaultstorageconfig'
-    properties: {
-      StorageModelType: vaultStorageType
-      CrossRegionRestoreFlag: enableCRR
-    }
+module rsv_backupStorageConfig 'backupStorageConfig/deploy.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-BackupStorageConfig'
+  params: {
+    recoveryVaultName: rsv.name
+    vaultStorageType: backupStorageConfig.vaultStorageType
+    enableCRR: backupStorageConfig.enableCRR
   }
-
-  resource rsv_protectionContainers 'protectionContainers@2016-12-01' = [for (protectionContainer, index) in protectionContainers: {
-    name: protectionContainer.Name
-    location: resourceGroup().location
-    properties: {
-      sourceResourceId: (empty(protectionContainer.sourceResourceId) ? json('null') : protectionContainer.sourceResourceId)
-      friendlyName: (empty(protectionContainer.friendlyName) ? json('null') : protectionContainer.friendlyName)
-      backupManagementType: (empty(protectionContainer.backupManagementType) ? json('null') : protectionContainer.backupManagementType)
-      containerType: (empty(protectionContainer.containerType) ? json('null') : protectionContainer.containerType)
-    }
-  }]
-
-  resource rsv_backupPolicies 'backupPolicies@2019-06-15' = [for (protectionPolicy, index) in backupPolicies: {
-    name: protectionPolicy.name
-    location: resourceGroup().location
-    properties: protectionPolicy.Properties
-  }]
+  dependsOn: [
```

Comment:
No need for `dependsOn `as the dependency is automatically get by using the `rsv.name` property above. 
Bicep magic 🧙

Reply:
Updated!